include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -177,23 +182,45 @@ class CSummedAreaTableImageFilter : public CMatchedSizeInOutImageFilterCommon, p
 
 					for (uint8_t channel = 0; channel < currentChannelCount; ++channel)
 					{ 
-						*(mainRowCache + channel) += decodeBuffer[channel];
-						*(mainColumnCache + channel) += decodeBuffer[channel];
+						*(mainRowCache + channel) += newReadBlockPos.x > 0 ? decodeBuffer[channel] : 0;
+						*(mainColumnCache + channel) += newReadBlockPos.y < trueExtent.y - 1 ? decodeBuffer[channel] : 0;
 
 						*(finalPixel + channel)
-						= (newReadBlockPos.x > 0 ? mainRowCache[channel] : 0)
-						+ (newReadBlockPos.y < trueExtent.y - 1 ? *(mainColumnCache + channel) : 0)
+						= mainRowCache[channel]
+						+ (newReadBlockPos.y < trueExtent.y - 1 ? *(mainColumnCache + channel) : 0) // gotta check it as well as above to work correctly
 						+ (!ExclusiveMode ? decodeBuffer[channel] : 0);
 					}
 
 					memcpy(scratchMemory + outDataAdressOffsetScratch, finalPixel, sizeof(decodeType) * currentChannelCount);
 
 					if (newReadBlockPos.x == commonExecuteData.oit->imageExtent.width - 1) // reset row cache when changing y
+					{
+						if (newReadBlockPos.y == 0) // sum values in (maxX, maxY, z)
+							memcpy(imageTotalCache + newReadBlockPos.z * currentChannelCount, finalPixel, sizeof(decodeType) * currentChannelCount);
+							
 						memset(mainRowCache, 0, sizeof(decodeType) * currentChannelCount);
+					}
 				};
 
 				CBasicImageFilterCommon::executePerRegion(commonExecuteData.inImg, sum, commonExecuteData.inRegions.begin(), commonExecuteData.inRegions.end(), clip);
 
+				auto normalizeScratch = [&]()
```

Comment:
what's this?

Reply:
we wanted something like `divideEverythingBy` where inputs are `imageTotal` (maxX, maxY) sum

Reply:
that needs to be an option

Reply:
I will return it via state

Reply:
okay, though I will return it via the filter